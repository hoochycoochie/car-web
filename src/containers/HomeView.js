import React, { useEffect } from "react";
import { Message } from "semantic-ui-react";
import { useDispatch, useSelector } from "react-redux";
import { fetchCars } from "../store/actions/car";

import { CarList } from "../components/CarList";

export default function HomeView() {
  const { data, error } = useSelector(state => state.cars);

  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(fetchCars());
  }, []);

  if (error) {
    return (
      <Message negative size="huge">
        {error.message}
      </Message>
    );
  }
  return <CarList cars={data && data.docs ? data.docs : []} />;
}
