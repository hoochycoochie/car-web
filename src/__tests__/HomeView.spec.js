import React from "react";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import configureStore from "redux-mock-store";
import HomeView from "../containers/HomeView";

const mockStore = configureStore([]);

describe("Testing HomeView container", () => {
  let store;
  let component;

  beforeEach(() => {
    store = mockStore({
      cars: []
    });

    store.dispatch = jest.fn();
    component = renderer.create(
      <Provider store={store}>
        <HomeView />
      </Provider>
    );
  });

  it("should render with the givent state from the redux store", () => {
    expect(component.toJSON()).toMatchSnapshot();
  });

  it(" should dispatch fetch user on componentDidMount", () => {
    expect(store.dispatch).toHaveBeenCalledTimes(0);
  });
});
