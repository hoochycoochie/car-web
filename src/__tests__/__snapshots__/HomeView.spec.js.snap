// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing HomeView container should render with the givent state from the redux store 1`] = `null`;
